openassessment/xblock/student_training_mixin.py

```diff
@@ -128,23 +128,46 @@ def training_path_and_context(self):
                 },
                 examples
             )
+            error_message = ""
             if example:
-                # EDUCATOR-1263: some entries may be in an incorrect format, load them anyways
-                if isinstance(example, dict) and isinstance(example['answer'], list) and isinstance(example['answer'][0], basestring):
-                    context['training_essay'] = create_submission_dict({'answer': {'parts': [{'text': example['answer'][0]}]}}, self.prompts)
-                else:
-                    context['training_essay'] = create_submission_dict({'answer': example['answer']}, self.prompts)
+                if isinstance(example, dict) and isinstance(example['answer'], list) and len(example['answer']) > 0 \
+                    and isinstance(example['answer'][0], basestring):
+                    context['training_essay'] = create_submission_dict(
+                        {'answer':
+                            {'parts':
+                                [{'text': example['answer'][0]}]
+                            }
+                        },
+                        self.prompts
+                    )
+                elif isinstance(example, dict) and isinstance(example['answer'], basestring):
```

Comment:
The second check is now explicitly looking for a dict containing a string, rather than the bare `else` used before.